src/Pages/components/Searchbar.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from 'react'
+import { v4 as uuid } from 'uuid'
+import styled from 'styled-components/macro'
+import useEvents from '../EventMain/useEvents'
+import Input from './common/Input'
+
+export default function Searchbar({
+  handleEventSuggestion,
+  handleEventSearch,
+}) {
+  const { eventArray } = useEvents()
+  const eventSuggestions = eventArray.map((event) => event.name)
+  const [filteredSuggestions, setFilteredSuggestions] = useState([])
+  const [eventNameInput, setEventNameInput] = useState('')
+
+  return (
+    <>
+      <EventSearchbar
+        placeholder="search..."
+        value={eventNameInput}
+        onKeyUp={handleInputKeyUp}
+        onChange={(e) => setEventNameInput(e.target.value)}
+      />
+      <SuggestionContainer>
+        <SuggestionList>
+          {filteredSuggestions.map((sug) => (
+            <SuggestionItems
+              key={uuid()}
+              onClick={() => handleEventSuggestionClick(sug)}
+            >
+              {getMarkedSuggestionText(sug)}
+            </SuggestionItems>
+          ))}
+        </SuggestionList>
+      </SuggestionContainer>
+    </>
+  )
+  function handleInputKeyUp(event) {
+    const inputVal = event.target.value
+    if (event.key === 'Enter') {
+      setEventNameInput('')
+      setFilteredSuggestions([])
+      handleEventSearch(inputVal)
+    } else if (inputVal.length < 3) {
+      setFilteredSuggestions([])
+    } else {
+      updateFilteredSuggestions()
+    }
+  }
+
+  function handleEventSuggestionClick(sug) {
+    setEventNameInput('')
+    setFilteredSuggestions([])
+    handleEventSuggestion(sug)
+  }
+
+  function updateFilteredSuggestions() {
+    if (eventNameInput.length < 3) return
+    setFilteredSuggestions(
+      eventSuggestions.filter((s) =>
+        s.toLowerCase().includes(eventNameInput.toLowerCase())
+      )
+    )
+  }
+
+  function getMarkedSuggestionText(suggestion) {
+    const searchIndex = suggestion
+      .toLowerCase()
+      .indexOf(eventNameInput.toLowerCase())
+
+    let firstPart = suggestion.slice(0, searchIndex)
```

Comment:
Do you need to use "let" here instead of "const"? I think it might work with const too since you don't reassign the variable.

Reply:
I will try, but thought it has to be let, will try again